packages/ember-template-compiler/lib/plugins/transform-with-as-to-hash.js

```diff
@@ -37,11 +37,17 @@ TransformWithAsToHash.prototype.transform = function TransformWithAsToHash_trans
 
   walker.visit(ast, function(node) {
     if (pluginContext.validate(node)) {
-
       if (node.program && node.program.blockParams.length) {
         throw new Error('You cannot use keyword (`{{with foo as bar}}`) and block params (`{{with foo as |bar|}}`) at the same time.');
       }
 
+      Ember.deprecate(
```

Comment:
I could also set a flag here and then do the actual deprecation in the `{{with}}` helper.